### ISSUE TITLE
Dashboard [Metrics GCP] Compute Overview - Data not properly filtered

### ISSUE THREAD

**Issue by Danouchka** [elastic/integrations#5007]
Hi

Regarding elastic-agent **CGP compute metrics** integration , all gcp compute metrics are searched in all metrics-* indices.
As a result, machines that are not necessarily in GCP are displayed on the dashboard **[Metrics GCP] Compute Overview**

All panels should be filtered with  **event.dataset:"gcp.compute"** in order to retrieve only the meaningful metrics 

Thank you

**Comment by endorama** [comment]
Hi @Danouchka, may you help me better understand this phrase:
> As a result, machines that are not necessarily in GCP are displayed on the dashboard

GCP Compute dashboard uses fields that are unique to the `gcp` package (and I would expect to the `gcp.compute` data stream too). Due to this, is not clear to me how machines not in GCP could be displayed on this dashboard. 

May you help me clarify what is happening, or what are you observing? Thanks

**Comment by tommyers-elastic** [comment]
i agree that this should not cause an issue with incorrect data being displayed (the panels query fields such as `gcp.compute.instance.cpu.usage.pct`), however it does have a performance impact. see https://github.com/elastic/integrations/issues/4936 for details - can we close this issue as a dup?

**Comment by endorama** [comment]
TLDR: I investigated this offline with @Danouchka, and the issue is not with the data but with the legend. I would not close this as duplicate of #4936 but would instead work on adding the dataset filter to all our dashboards.

# Long investigation:
Dan is collecting data for GCP instances in 2 different projects. 
In Project A he is collecting data through `system` integration (no `gcp` integration is used).
In Project B he is collecting data through `gcp.compute` integrations.

The GCP Compute visualizations show instances belonging to Project A, albeit the integration is not collecting any data from Project A. As there is no `gcp.compute.*` collection for those instances, the graph is empty. The instance names are displayed in the legend.

This is confusing, as in this case the instances in Project B where not even shown, only instances from Project A with empty data points.

# Why is this happening?

The issue is not with the collected data, but with the aggregation we perform on `cloud.instance.name`. This field is part of ECS and can be filled by any other integration.
The result is that we are storing data that contains `cloud.instance.name` but no `gcp.compute.*` fields, resulting in a legend displaying instances for which the `gcp.compute` data stream is not collecting data.

# Conclusion

The root cause is the same, but I would consider this a bug, not just a performance concern. I would also trigger an investigation in all our other dashboards, as I suspect we are not filtering for dataset and this could affect other dashboards and visualizations.

**Comment by Danouchka** [comment]
Fixed as dashboard filtered on event.dataset: gcp.compute
